### ISSUE TITLE
求教，怎么训练能得到可用的模型

### ISSUE THREAD

**Issue by Runist** [WHUHLX/CATS#12]
我是直接用了您的代码，然后因为主页的数据集挂了，所以我在kaggle上下载了hed-bsds的数据，数量也是28800张，和您提供的lst里面的文件名也一样，应该是一样的数据集。但我按照您的配置去用vgg训练，30个epoch后得到的模型，效果很差啊，和您提供的权重预测效果不一样。
![image](https://user-images.githubusercontent.com/38599727/211229329-79bf5a02-4849-48f8-aeb0-1fdf5d6a9057.png)

**Comment by Runist** [comment]
求教，要怎么样训练可以得到可用的模型

**Comment by PlanetaryScience3510** [comment]
> 我是直接用了您的代码，然后因为主页的数据集挂了，所以我在kaggle上下载了hed-bsds的数据，数量也是28800张，和您提供的lst里面的文件名也一样，应该是一样的数据集。但我按照您的配置去用vgg训练，30个epoch后得到的模型，效果很差啊，和您提供的权重预测效果不一样。 ![image](https://user-images.githubusercontent.com/38599727/211229329-79bf5a02-4849-48f8-aeb0-1fdf5d6a9057.png)

我也在训练这个数据，但是发现loss 不下降，请问您遇到过这个问题吗

**Comment by yuanxuanS** [comment]
我感觉代码里loss_wce权重这块有问题：
def tracingloss(prediction, label, tex_factor=0., bdr_factor=0., balanced_w=1.1):
    label = label.float()
    prediction = prediction.float()
    with torch.no_grad():
        mask = label.clone()

        num_positive = torch.sum((mask==1).float()).float()
        num_negative = torch.sum((mask==0).float()).float()
        beta = num_negative / (num_positive + num_negative)     # 论文中的 1 - alpha
        mask[mask == 1] = beta
        mask[mask == 0] = balanced_w * (1 - beta)
        mask[mask == 2] = 0

mask ==0 和 mask == 1 的部分是不是反了？

**Comment by yuanxuanS** [comment]
> 我感觉代码里loss_wce权重这块有问题： def tracingloss(prediction, label, tex_factor=0., bdr_factor=0., balanced_w=1.1): label = label.float() prediction = prediction.float() with torch.no_grad(): mask = label.clone()
> 
> ```
>     num_positive = torch.sum((mask==1).float()).float()
>     num_negative = torch.sum((mask==0).float()).float()
>     beta = num_negative / (num_positive + num_negative)     # 论文中的 1 - alpha
>     mask[mask == 1] = beta
>     mask[mask == 0] = balanced_w * (1 - beta)
>     mask[mask == 2] = 0
> ```
> 
> mask ==0 和 mask == 1 的部分是不是反了？

应该为
   mask[mask == 1] = 1- beta
    mask[mask == 0] = balanced_w *  beta

**Comment by yuanxuanS** [comment]
> > 我感觉代码里loss_wce权重这块有问题： def tracingloss(prediction, label, tex_factor=0., bdr_factor=0., balanced_w=1.1): label = label.float() prediction = prediction.float() with torch.no_grad(): mask = label.clone()
> > ```
> >     num_positive = torch.sum((mask==1).float()).float()
> >     num_negative = torch.sum((mask==0).float()).float()
> >     beta = num_negative / (num_positive + num_negative)     # 论文中的 1 - alpha
> >     mask[mask == 1] = beta
> >     mask[mask == 0] = balanced_w * (1 - beta)
> >     mask[mask == 2] = 0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > mask ==0 和 mask == 1 的部分是不是反了？
> 
> 应该为 mask[mask == 1] = 1- beta mask[mask == 0] = balanced_w * beta
期待作者回复

**Comment by CookiC** [comment]
@yuanxuanS 这篇文章和BDCN感觉都有点问题，我都没复现出来。BDCN和这篇文章你后来复现出来了吗？
